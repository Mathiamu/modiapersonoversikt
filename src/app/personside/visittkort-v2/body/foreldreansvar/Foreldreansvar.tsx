import * as React from 'react';
import { VisittkortGruppe } from '../VisittkortStyles';
import VisittkortElement from '../VisittkortElement';
import { Normaltekst } from 'nav-frontend-typografi';
import Infotegn from '../../../../../svg/Info';
import { Foreldreansvar, NavnOgIdent } from '../../PersondataDomain';
import { hentNavn } from '../../visittkort-utils';

interface Props {
    foreldreansvar: Foreldreansvar[];
}

function kombinerNavnOgIdent(personInfo: NavnOgIdent | null): string | null {
    if (!personInfo) {
        return null;
    }
    const navn = hentNavn(personInfo.navn);
    return personInfo.navn ? `${navn} (${personInfo.ident})` : navn;
}

function ForeldreansvarElement(props: { foreldreansvar: Foreldreansvar }) {
    const { foreldreansvar } = props;
    const ansvarlig = kombinerNavnOgIdent(foreldreansvar.ansvarlig);
    const ansvarsubject = kombinerNavnOgIdent(foreldreansvar.ansvarsubject);

    return (
        <VisittkortElement>
            <Normaltekst>Ansvar: {foreldreansvar.ansvar}</Normaltekst>
            {ansvarlig && <Normaltekst>Ansvarlig: {ansvarlig}</Normaltekst>}
            {ansvarsubject && <Normaltekst>Gjelder for: {ansvarsubject}</Normaltekst>}
        </VisittkortElement>
    );
}

function ForendreansvarWrapper({ foreldreansvar }: Props) {
    if (foreldreansvar.isEmpty()) {
        return null;
    }

    return (
        <VisittkortGruppe ikon={<Infotegn />} tittel="Foreldreansvar">
            {foreldreansvar.map((foreldreansvar, index) => (
                <ForeldreansvarElement key={index} foreldreansvar={foreldreansvar} />
            ))}
        </VisittkortGruppe>
    );
}
export default ForendreansvarWrapper;
