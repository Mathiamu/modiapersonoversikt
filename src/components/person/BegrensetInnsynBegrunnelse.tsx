import * as React from 'react';
import { BegrensetInnsynTyper } from '../../models/person/person';

interface Props {
    begrunnelseType: BegrensetInnsynTyper;
}

function BegrensetInnsynBegrunnelse({begrunnelseType}: Props) {
    switch (begrunnelseType) {
        case BegrensetInnsynTyper.Kode6:
            return <>Bruker har kode 6 adresse</>;
        case BegrensetInnsynTyper.Kode7:
            return <>Bruker har kode 7 adresse</>;
        case BegrensetInnsynTyper.EgenAnsatt:
            return <>Bruker faller inn under begrensning for egen ansatt</>;
        default:
            return <>Ukjent sikkerhetsbegrensing</>;
    }
}

export default BegrensetInnsynBegrunnelse;