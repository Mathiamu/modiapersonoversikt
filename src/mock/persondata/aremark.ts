import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SikkerhetstiltakType,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';

export const barnAremark: ForelderBarnRelasjon[] = [
    {
        ident: '12345678910',
        rolle: ForelderBarnRelasjonRolle.BARN,
        navn: [
            {
                fornavn: 'BARN1',
                mellomnavn: null,
                etternavn: 'BARNESEN'
            }
        ],
        fodselsdato: ['2008-03-15' as LocalDate],
        alder: 13,
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.UGRADERT,
                beskrivelse: 'UGRADERT'
            }
        ],
        bostedAdresse: [
            {
                linje1: 'Islandsgate 49',
                linje2: '7693 Hovedøya',
                linje3: 'Norge'
            }
        ],
        personstatus: [
            {
                kode: PersonStatus.BOSATT,
                beskrivelse: 'BOSATT'
            }
        ]
    }
];

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge'
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum'
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldigFraOgMed: '1980-10-10' as LocalDate,
            gyldigTilOgMed: null,
            erHistorisk: false
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Ingen'
        }
    ],
    sikkerhetstiltak: [
        {
            type: SikkerhetstiltakType.TFUS,
            gyldigFraOgMed: '2005-02-13' as LocalDate,
            gyldigTilOgMed: '2030-02-15' as LocalDate
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate
        }
    ],
    foreldreansvar: [
        {
            ansvar: 'felles',
            ansvarlig: {
                fornavn: 'Test',
                etternavn: 'Testesen',
                mellomnavn: null
            },
            ansvarsubject: {
                fornavn: 'Barn',
                etternavn: 'Barnesen',
                mellomnavn: null
            }
        }
    ],
    deltBosted: [
        {
            startdatoForKontrakt: '2000-10-10' as LocalDate,
            sluttdatoForKontrakt: '2005-05-07' as LocalDate,
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omraade: ['*'],
            gyldigFraOgMed: '2015-01-01' as LocalDate,
            gyldigTilOgMed: '2017-12-12' as LocalDate
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'voksen',
            omfang: 'personligeOgOekonomiskeInteresser',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetstidspunkt: '2016-03-27' as LocalDate,
            opphoerstidspunkt: '2017-06-30' as LocalDate
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'retningsnummer er norsk'
            },
            identifikator: '99009900',
            sistEndret: '2017-10-10T10:15:30' as LocalDateTime,
            sistEndretAv: 'TESTFAMILIEN AREMARK',
            prioritet: 1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '90000000',
            sistOppdatert: null,
            sistVerifisert: null
        }
    },
    bankkonto: {
        kontonummer: '1234567890',
        banknavn: 'DNB ASA',
        sistEndret: '2006-03-15T00:00:00' as LocalDateTime,
        sistEndretAv: '1010800 BD03',

        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: barnAremark
};
