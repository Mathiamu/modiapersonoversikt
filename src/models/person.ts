import { Kodeverk } from './kodeverk';

export interface Person {
    navn: Navn;
    kjønn: Kjønn;
    geografiskTilknytning?: string;
    fødselsnummer: string;
    alder: number;
    diskresjonskode?: string;
    bankkonto?: Bankkonto;
    statsborgerskap?: string;
    personstatus: Bostatus;
    sivilstand: Kodeverk;
    familierelasjoner: Familierelasjon[];
}

export interface Navn {
    sammensatt: string;
    fornavn: string;
    mellomnavn: string;
    etternavn: string;
}

export interface Bankkonto {
    erNorskKonto: boolean;
    kontonummer: number;
    bank: string;
    sistEndret: string;
    sistEndretAv: string;
}

export interface Familierelasjon {
    harSammeBosted: boolean;
    rolle: Relasjonstype;
    tilPerson: {
        navn: Navn;
        alder: number;
        fødselsnummer: string;
        personstatus: Bostatus;
        kjonn: Kjønn
    };
}

export interface Bostatus {
    dødsdato?: string;
    bostatus?: string;
}

export enum BostatusTyper {
    Død = 'DØD',
    Utvandret = 'UTVA'
}

export enum Relasjonstype {
    Barn = 'BARN',
    Samboer = 'SAMBOER',
    Partner = 'PARTNER',
    Ektefelle = 'EKTE',
    Gift = 'GIFT'
}

export enum Sivilstand {
    Gift = 'GIFT',
    Ugift = 'UGIFT'
}

export enum Kjønn {
    Mann = 'M',
    Kvinne = 'K'
}

export function erDød(personstatus: Bostatus) {
    return personstatus.dødsdato || personstatus.bostatus === BostatusTyper.Død;
}

export function getBarnUnder21(familierelasjoner: Familierelasjon[]) {
    return familierelasjoner
        .filter(relasjon => relasjon.rolle === Relasjonstype.Barn)
        .filter(relasjon => relasjon.tilPerson.alder <= 21);
}

export function getPartner(person: Person) {
    const aktuelleRelasjoner = [
        Relasjonstype.Ektefelle,
        Relasjonstype.Gift,
        Relasjonstype.Partner,
        Relasjonstype.Samboer
    ];
    return person.familierelasjoner.find(relasjon => aktuelleRelasjoner.includes(relasjon.rolle));
}
