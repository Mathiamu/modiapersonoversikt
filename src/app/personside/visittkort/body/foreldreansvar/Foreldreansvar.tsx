import * as React from 'react';
import { VisittkortGruppe } from '../VisittkortStyles';
import { Foreldreansvar } from '../../../../../models/foreldreansvar/foreldreansvar';
import VisittkortElement from '../VisittkortElement';
import { Normaltekst } from 'nav-frontend-typografi';
import Infotegn from '../../../../../svg/Info';

function ForeldreansvarElement(props: { foreldreansvar: Foreldreansvar }) {
    const ansvarssubjekt =
        props.foreldreansvar.ansvarssubjekt.sammensatt.length > 0
            ? props.foreldreansvar.ansvarssubjekt.sammensatt
            : 'Navn ikke tilgjengelig';
    return (
        <VisittkortElement>
            <Normaltekst>
                {props.foreldreansvar.ansvarlig?.sammensatt ?? 'Navn ikke tilgjengelig'}({props.foreldreansvar.ansvar})
            </Normaltekst>
            <Normaltekst>Gjelder for: {ansvarssubjekt}</Normaltekst>
        </VisittkortElement>
    );
}

function ForendreansvarWrapper(props: { foreldreansvar?: Foreldreansvar[] }) {
    if (!props.foreldreansvar || props.foreldreansvar.length === 0) {
        return null;
    }
    const foreldreansvarElementer = props.foreldreansvar.map((foreldreansvar, index) => (
        <ForeldreansvarElement key={index} foreldreansvar={foreldreansvar} />
    ));

    return (
        <VisittkortGruppe ikon={<Infotegn />} tittel="Foreldreansvar">
            {foreldreansvarElementer}
        </VisittkortGruppe>
    );
}
export default ForendreansvarWrapper;
