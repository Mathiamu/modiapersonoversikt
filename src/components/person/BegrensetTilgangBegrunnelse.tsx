import * as React from 'react';
import { BegrensetTilgangTyper } from '../../models/person/person';

interface Props {
    begrunnelseType: BegrensetTilgangTyper;
}

function BegrensetTilgangBegrunnelse({begrunnelseType}: Props) {
    switch (begrunnelseType) {
        case BegrensetTilgangTyper.Kode6:
            return <>Bruker har strengt fortrolig adresse</>;
        case BegrensetTilgangTyper.Kode7:
            return <>Bruker har fortrolig adresse</>;
        case BegrensetTilgangTyper.EgenAnsatt:
            return <>Bruker faller inn under begrensning for egen ansatt</>;
        case BegrensetTilgangTyper.DefaultFraBackEnd:
            return <>Ukjent sikkerhetsbegrensing</>;
        default:
            return <>Feil sikkerhetsbegrensing</>;
    }
}

export default BegrensetTilgangBegrunnelse;