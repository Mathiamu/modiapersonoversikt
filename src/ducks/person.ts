import { Person } from '../models/person';
import { Action, Dispatch } from 'redux';
import { FetchError, FetchSuccess, RestActions, STATUS } from './utils';

export interface PersonState {
    data: Person;
    status: STATUS;
    error: string | null;
}

const mockPerson = {
    fornavn: 'Aremark',
    etternavn: 'Testfamilien',
    fodselsnummer: '10108000398'
};

const reducerName = 'person/';

const actionNames = {
    PENDING: reducerName + STATUS.PENDING,
    OK: reducerName + STATUS.OK,
    ERROR: reducerName + STATUS.ERROR
};

const initialState: PersonState = {
    data: mockPerson,
    status: STATUS.NOT_STARTED,
    error: null
};

function mockFetch(fodselsnummer: string) {
    return new Promise((resolve, reject) => {
        setTimeout(() => { resolve(mockPerson); }, 2500);
    });
}

export function hentPerson(fodselsnummer: string) {
    return (dispatch: Dispatch<Action>) => {
        dispatch({ type: actionNames.PENDING});
        return mockFetch(fodselsnummer)
            .then((person) => {
                dispatch({type: actionNames.OK, data: person});
                return person;
            })
            .catch((err) => {
                dispatch({type: actionNames.ERROR, error: err});
            });
    };
}

export default function reducer(state: PersonState = initialState, action: RestActions<Person>): PersonState {
    switch (action.type) {
        case actionNames.PENDING:
            return {...state, status: STATUS.PENDING };
        case actionNames.OK:
            return {...state, status: STATUS.OK, data: (<FetchSuccess<Person>> action).data};
        case actionNames.ERROR:
            return {...state, status: STATUS.ERROR, error: (<FetchError> action).error};
        default:
            return state;
    }
}