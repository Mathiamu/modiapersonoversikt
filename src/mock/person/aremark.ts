import { Diskresjonskoder } from '../../constants';
import { Person } from '../../models/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: 'M',
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    status: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: 'GIFT',
        beskrivelse: 'Gift'
    },
    familierelasjoner: []
};