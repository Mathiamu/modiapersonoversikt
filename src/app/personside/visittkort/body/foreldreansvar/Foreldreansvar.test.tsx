import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../../test/Testprovider';
import { aremark } from '../../../../../mock/person/aremark';
import Foreldreansvar from './Foreldreansvar';

test('viser foreldreansvar', () => {
    const visittkortbody = renderer.create(
        <TestProvider>
            <Foreldreansvar foreldreansvar={aremark.foreldreansvar} />
        </TestProvider>
    );

    expect(visittkortbody.toJSON()).toMatchSnapshot();
});
