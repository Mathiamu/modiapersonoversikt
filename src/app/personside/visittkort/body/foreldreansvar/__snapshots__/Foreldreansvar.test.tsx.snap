// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser foreldreansvar 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: absolute;
  left: -3.125rem;
  width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > svg {
  height: 1.5rem;
  width: auto;
}

.c2 {
  margin-bottom: 1rem;
}

.c3 {
  margin: 10px 0 20px 0;
}

.c3:last-child {
  margin: 10px 0 0 0;
}

<section>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <svg
        fill="#0067C5"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          clip-rule="evenodd"
          d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0zm0 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zM9 19v-2h2v-5H9v-2h4v7h2v2H9zm3-14a1.5 1.5 0 110 3 1.5 1.5 0 010-3z"
          fill="#0067C5"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <h3
      className="typo-undertittel"
    >
      Foreldreansvar
    </h3>
  </div>
  <div
    className="c2"
  />
  <div
    className="c3"
  >
    <div
      className="c0"
    >
      
    </div>
    <p
      className="typo-normal"
    >
      Ola Normann
      (
      felles
      )
    </p>
  </div>
  <div
    className="c3"
  >
    <div
      className="c0"
    >
      
    </div>
    <p
      className="typo-normal"
    >
      Petter Pettersen
      (
      far
      )
    </p>
  </div>
</section>
`;
